share/functions/funced.fish

```diff
@@ -122,6 +130,26 @@ function funced --description 'Edit function definition'
                     continue
                 end
                 echo (_ "Cancelled function editing")
+            else if test -n "$writepath"
+                if set -q _flag_save
+                    # try to write the file back
+                    # cp preserves existing permissions, though it might overwrite the owner
+                    if not cp $tmpname "$writepath" 2>&1
+                        echo (_ "Saving to original location failed; saving to user configuration instead.")
+                        set writepath $__fish_config_dir/functions/(basename "$writepath")
+                        if not cp $tmpname "$writepath"
+                            echo (_ "Saving to user configuration failed. Changes will be lost when fish is closed.")
+                        else
+                            # read it back again - this ensures that the output of `functions --details` is correct
+                            source "$writepath"
+                        end
+                    else
+                        # read it back again - this ensures that the output of `functions --details` is correct
```

Comment:
I suggest adding a confirmation message here, perhaps:

    `Function saved to $writepath`

Reply:
Good idea.